algorithm/diagonal-difference/solutions/diagonalDifference.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Time complexity - O(|arr|)
+ */
+function diagonalDifference(arr) {
+  let principle = 0;
+  let secondary = 0;
+  for (let i = 0; i < arr.length; i++) {
+    principle += arr[i][i];
+    secondary += arr[i][arr.length - i - 1];
+  }
```

Comment:
No need to use 2 variables. Use only variable, sum up primary diagonals, minus secondary diagonals.